### ISSUE TITLE
Video with srcObject never loads on Safari

### ISSUE THREAD

**Issue by brunoimbrizi** [dmnsgn/async-preloader#25]
After [Use srcObject for HTMLMediaElement](https://github.com/dmnsgn/async-preloader/commit/e59f4ae574b98758e87fcd4cf48f42870108a6d8) the issues with [videos never loading on Safari](https://github.com/dmnsgn/async-preloader/issues/9) are back.

Example:
```
import AsyncPreloader from "async-preloader";

const items = [{ "id": "myVideoFile", "src": "assets/video.mp4" }];
const pItems = AsyncPreloader.loadItems(items);
```

- MacOS Mojave Safari 12
- MacOS High Sierra Safari 11.1
- iPhone XS iOS 12 Safari
- iPhone 8 iOS 11 Safari

> [Error] Not allowed to load local resource: blob:blobinternal:///f036178f-fd56-4c6a-9cb5-29b27b9b11a4

- MacOS Sierra Safari 10
- iPhone 7 iOS 10 Safari

> Preloader fails silently. No errors.

--

Placing logs inside the `try...catch` block in [loadVideo](https://github.com/dmnsgn/async-preloader/blob/master/src/index.ts#L226) shows that in all of the scenarios above the execution never goes inside `catch`, hence stopping at `video.srcObject = data;`

By commenting out the `try...catch` block and leaving only `video.src = URL.createObjectURL(data);` the results change:

- MacOS Mojave Safari 12
- MacOS High Sierra Safari 11.1
- iPhone XS iOS 12 Safari
- iPhone 8 iOS 11 Safari

> Preloader works. `canplaythrough` is fired.

- MacOS Sierra Safari 10
- iPhone 7 iOS 10 Safari

> `error` is fired [here](https://github.com/dmnsgn/async-preloader/blob/master/src/index.ts#L241) and promise is rejected.

--

Safari's compatibility with `srcObject` is listed as unknown on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/srcObject).

**Comment by dmnsgn** [comment]
Should be fixed in latest version (`4.5.0`)

**Comment by dmnsgn** [comment]
Closing as I am forcing `video.src` when Safari is detected in `navigator.userAgent`:
https://github.com/dmnsgn/async-preloader/blob/cb7a477b03b540a6d6601f4568a0857899fc79ca/src/index.ts#L248-L253
